examples/plot_vgm_error.py

```diff
@@ -0,0 +1,256 @@
+# and normalized median absolute deviation. For this example, we do not account for the non-stationarity in elevation
+# measurement errors quantified in :ref:`sphx_glr_auto_examples_plot_nonstationary_error.py`.
+print('STD: {:.2f}'.format(np.nanstd(dh.data))+' meters.')
+print('NMAD: {:.2f}'.format(xdem.spatialstats.nmad(dh.data))+' meters.')
+
+# %%
+# The two measures of dispersion are quite similar showing that, on average, there is a small influence of outliers on the
+# elevation differences. The per-pixel precision is about :math:`\pm` 2.5 meters.
+# **Does this mean that every pixel has an independent measurement error of** :math:`\pm` **2.5 meters?**
+# Let's plot the elevation differences to visually check the quality of the data.
+plt.figure(figsize=(8, 5))
+plt_extent = [
+    dh.bounds.left,
+    dh.bounds.right,
+    dh.bounds.bottom,
+    dh.bounds.top,
+]
+plt.imshow(dh.data.squeeze(), cmap="RdYlBu", vmin=-4, vmax=4, extent=plt_extent)
+cbar = plt.colorbar()
+cbar.set_label('Elevation differences (m)')
+plt.show()
+
+
+# %%
+# We clearly see that the residual elevation differences on stable terrain are not random. The positive and negative
+# differences (blue and red, respectively) appear correlated over large distances.
+#
+# Conclusion: **These correlated errors need to be quantified to reliably estimate elevation measurement errors for a
+# sum, or average of elevation differences samples for a specific surface area**.
+
+# %%
+# Additionally, we notice that the elevation differences are still polluted by unrealistically large elevation
+# differences near glaciers, probably because the glacier inventory is more recent than the data, and the outlines are too small.
+# To remedy this, we filter large elevation differences outside 4 NMAD.
+dh.data[np.abs(dh.data) > 4 * xdem.spatialstats.nmad(dh.data)] = np.nan
+
+# %%
+# We plot the elevation differences after filtering to check that we successively removed the reminaing glacier signals.
+plt.figure(figsize=(8, 5))
+plt_extent = [
+    dh.bounds.left,
+    dh.bounds.right,
+    dh.bounds.bottom,
+    dh.bounds.top,
+]
+plt.imshow(dh.data.squeeze(), cmap="RdYlBu", vmin=-4, vmax=4, extent=plt_extent)
+cbar = plt.colorbar()
+cbar.set_label('Elevation differences (m)')
+plt.show()
+
+# %%
+# To quantify the spatial correlation of the data, we sample an empirical variogram.
+# The empirical variogram describes the variance between the elevation differences of pairs of pixels depending on their
+# distance. This distance between pairs of pixels if referred to as spatial lag.
+# To perform this procedure effectively, we use improved methods that provide efficient pairwise sampling methods for
+# large grid data in `scikit-gstat <https://mmaelicke.github.io/scikit-gstat/index.html>`_, which are encapsulated
+# conveniently by :func:`xdem.spatialstats.sample_multirange_variogram`:
+df = xdem.spatialstats.sample_multirange_variogram(
```

Comment:
As discussed in PR #175, the functions in spatial_stats could be modified to accept either an array or a Raster.

Reply:
I think it will be more practical if we do a specific PR for this, as specified in #178